kong/db/schema/init.lua

```diff
@@ -1128,6 +1132,44 @@ local function resolve_field(self, k, field, subschema)
 end
 
 
+local function get_json_schema(field, field_name, input)
+  local json_schema = field.json_schema
+
+  local schema = json_schema.inline
+  if schema then
+    return schema
+  end
+
+  local parent_key = json_schema.parent_subschema_key
+  local subschema_key = input[parent_key]
+
+  if subschema_key then
+    local schema_name = json_schema.namespace .. "/" .. subschema_key
+    schema = json.get_schema(schema_name) or json_schema.default
+
+    if schema then
+      return schema
+
+    elseif not json_schema.optional then
+      return validation_errors.JSON_SCHEMA_NOT_FOUND:format(schema_name)
+    end
+
+  elseif not json_schema.optional then
+    return validation_errors.JSON_PARENT_KEY_MISSING:format(field_name, parent_key)
```

Comment:
```suggestion
    return nil, validation_errors.JSON_PARENT_KEY_MISSING:format(field_name, parent_key)
```

Reply:
I updated some type definitions and updated the json functions to use more idiomatic Lua return values.